runtime/permissions.rs

```diff
@@ -63,515 +65,528 @@ impl Default for PermissionState {
 
 #[derive(Clone, Debug, Default, Deserialize, PartialEq)]
 pub struct UnaryPermission<T: Eq + Hash> {
-  pub global_state: PermissionState,
+  #[serde(skip)]
+  pub name: String,
+  #[serde(skip)]
+  pub description: String,
+  pub state: PermissionState,
```

Comment:
What's the reason for this rename? This should be rigorously named `global_state`.
```suggestion
  pub global_state: PermissionState,
```
It doesn't make sense for `unary_permission.state` to be `PermissionState::Prompt` while `unary_permission.query(foo)` can return `PermissionState::Granted`.

Reply:
my bad, that was a mishap on my side